### PR TITLE
fix(spec): inline anonymous struct literals instead of emitting a dangling $ref

### PR DESCRIPTION
**Pull request by ehabterra**
## Problem

Generating a spec for code with a struct field typed `[]struct{...}` (a slice of an **anonymous** struct) produced an invalid OpenAPI document. ReDoc rejects it with:

```
Invalid reference token: github_com_..._http_struct{AssetID-string-"json:\"asset_id\"";-IsRequired-bool-...}
```

The emitted `$ref` is **dangling** (the component is never defined) and its name contains `{`, `}`, `"`, `;` — illegal characters in a JSON-pointer reference token.

## Root cause

An anonymous struct field type reaches the spec layer as its raw `go/types` `String()` form (`struct{AssetID string ...}`). `mapGoTypeToOpenAPISchema` had no case for a literal `struct{...}`, so it fell through to the "unknown custom type" branch and built a `$ref` to it — braces and all. The metadata layer already documents this gap (`metadata.go`: *"Field types like `[]struct{...}` or `*struct{...}` fall through to the string path"*).

## Fix (`internal/spec/mapper.go`)

- **Inline the literal.** `schemaFromAnonStructLiteral` parses the `struct{...}` form with `go/parser` and produces a proper inline object schema, honoring `json` tags. A guard in `mapGoTypeToOpenAPISchema` (after pointer/slice/map wrappers are peeled) returns it instead of a `$ref`. Result:
  ```yaml
  entries:
    type: array
    items:
      type: object
      properties:
        asset_id: { type: string }
        is_required: { type: boolean }
  ```
- **Never promote to a component.** `canAddRefSchemaForType` now rejects any `struct{...}` literal, so a nameless anonymous struct can't become a dangling, invalidly-named component anywhere (including via `shouldPromoteToComponent` and the recursion/cycle guards).

## Tests

`anon_struct_literal_test.go` covers bare / pointer / slice / package-prefixed forms. Each asserts the struct is inlined as an object with the correct properties and that **no component is ever registered under a braced name**. Full `internal/spec` + `generator` suites, `go vet`, and `gofmt` are clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Anonymous `struct{...}` Go types are now reliably inlined as OpenAPI object/array schemas, avoiding invalid `$ref` outputs.
  * JSON tag handling is corrected: only exact `json:"-"` omits fields; unexported fields are skipped while property names and nested types are preserved.

* **Tests**
  * Expanded coverage for anonymous struct inlining, including import-qualified field types, nested anonymous structs, and additional JSON tag omission edge cases.

### CODE REVIEW

File: internal/spec/mapper.go [hunk 1]
```diff
@@ -2171,10 +2187,82 @@ func canAddRefSchemaForType(key string) bool {
...
+			propName := n.Name
+			if jsonName != "" {
+				propName = jsonName
+			}
+			if schema.Properties == nil {
+				schema.Properties = map[string]*Schema{}
+			}
+			schema.Properties[propName] = fieldSchema
+		}
+	}
```
[the remark this answers is not in the document]
Comment (ehabterra) [reply internal/spec/mapper.go]: Fixed in 9999ca9 — confirmed real. Both `json:"-"` and unexported fields are never serialized by encoding/json, so they're now skipped before being added to the inline object schema (the whole field is skipped when no name is exported, avoiding orphan component schemas for their types).

On consistency: the named-struct path (`generateStructSchema`) had the same gap, so I applied identical filtering there via a shared `jsonFieldOmitted` helper + `ast.IsExported`. The full suite — including the generator fixture comparisons — stays green, so no existing fixture relied on emitting these non-serialized fields. Added tests covering both `json:"-"` and unexported fields.

File: internal/spec/mapper.go [hunk 2]
```diff
@@ -1171,6 +1180,18 @@ func getStringFromPool(meta *metadata.Metadata, idx int) string {
...
+// jsonFieldOmitted reports whether a struct field with the given tag is
+// excluded from JSON serialization entirely via a `json:"-"` tag. Note the
+// `json:"-,"` form names a field literally "-" and is NOT omitted.
+func jsonFieldOmitted(tag string) bool {
+	if !strings.Contains(tag, "json:") {
+		return false
+	}
+	parts := strings.SplitN(tag, "json:", 2)
+	jsonPart := strings.Trim(strings.Split(parts[1], " ")[0], "\"")
+	return jsonPart == "-"
```
[the remark this answers is not in the document]
Comment (ehabterra) [reply internal/spec/mapper.go]: Fixed in 97c9005 — confirmed real. `jsonFieldOmitted` now uses `reflect.StructTag(tag).Lookup("json")` for an exact-key match, so `myjson:"-"` no longer drops a field. Added a table test covering `json:"-"` (omit), `json:"-,"` (field literally named "-", kept), `myjson:"-"` (kept), and combined tags.

**Comment by ehabterra** [comment]
## Review round addressed (commit 97c9005)

**1. `json` tag exact-key lookup** — `jsonFieldOmitted` now uses `reflect.StructTag(tag).Lookup("json")` so unrelated keys like `myjson:"-"` no longer drop fields.

**2. ParseExpr can't handle import-path-qualified field types** (the AI-agent prompt) — confirmed real: `go/types` renders field types with full import paths (e.g. `github.com/google/uuid.UUID`), which aren't valid Go expressions, so `parser.ParseExpr` rejected the whole anonymous struct and dropped **every** field. Replaced `go/parser` with a small bracket/quote-aware splitter (`anonStructBody` + `splitTopLevel` + `parseAnonField`) that hands each field-type string straight to `mapGoTypeToOpenAPISchema`, which already resolves import paths, slices, maps, pointers, and nested anonymous structs. Added tests for import-qualified + nested-anon fields.

### Validation
Ran `scripts/compare-spec.sh -v 52` across all 38 projects. These changes produce a **byte-identical missing-set vs the previous commit** (zero regressions). The remaining v52 diffs are all pre-existing and not from this PR:
- Correct `json:"-"` drops — e.g. previously-leaked `Password` and `MediaStorageKey` fields are now (correctly) absent.
- The anonymous struct that's now inlined instead of emitting the invalid braced $ref (the original bug).
- External-project source drift (e.g. `CreateAddressBookRequest` fields changed in the source since the snapshot).
- A `DELETE /api/user/{id}` 400-response difference that also reproduces on `origin/main` (pre-existing).
